src/cls/IPM/Main.cls

```diff
@@ -2867,6 +2916,44 @@ ClassMethod EnableIPM(ByRef pCommandInfo)
 	}
 }
 
+ClassMethod History(ByRef pCommandInfo)
+{
+	Set action = $Get(pCommandInfo("parameters", "action"))
+
+	If (action = "find") || (action = "") {
+		Set globally = $$$HasModifier(pCommandInfo, "globally")
+		Set ascend = $$$HasModifier(pCommandInfo, "ascend")
+		Set limit = $$$GetModifier(pCommandInfo, "limit")
+
+		Merge filter = pCommandInfo("data", "zpm")
+		Set mthd = $Select(globally: "GetHistoryGlobally", 1: "GetHistory")
+		Set rs = $Classmethod("%IPM.General.History", mthd, .filter, ascend, limit)
```

Comment:
Nit: Rather than using dynamic method dispatch which loses intellisense, could have GetHistory() call GetHistoryGlobally() if it is a global search based on a method arg. Same with below